src/app.jsx

```diff
@@ -41,16 +41,24 @@ export class App extends ReactiveComponent {
 			<AddressBookSegment />
 			<Divider hidden />
 			<FundingSegment />
-			<Divider hidden />
-			<StakingSegment />
+			{runtime.staking != null &&
+			<div>
+			 <Divider hidden />
+			 <StakingSegment />
+			</div>
+			}
 			<Divider hidden />
 			<UpgradeSegment />
 			<Divider hidden />
 			<PokeSegment />
 			<Divider hidden />
 			<ParachainSegment />
-			<Divider hidden />
-			<ValidationSegment />
+			{runtime.staking != null &&
```

Comment:
I think this wrapper should live at the segment level right?

Also seems like if you do it for one module, you gotta do it for all of them XD

Reply:
> I think this wrapper should live at the segment level right?
What do you mean? Which wrapper, the div?

> Also seems like if you do it for one module, you gotta do it for all of them XD
I wasn't clear about that. For now I just made the necessary changes so that things work :)
Is it useful to do it for others? When/how are modules available or not?

Reply:
Modules are included by the user through the `construct_runtime!` macro. We cannot know ahead of time what modules a user might have in their runtime. I think the intention here, which is a good one, is to only show the parts of the Substrate UI when the correct module exists in the runtime.

> What do you mean? Which wrapper, the div?

Ideally I think this check for the staking module would happen within the `validation segment` itself? Or is that not possible? Would need to look more into it, but trust your judgement on the solution here.

Reply:
Thanks!
The original intention was to disable the whole segment if it's essentially empty when the module is absent. I will double check that.

Reply:
Updated according to your comments (see updated PR initial comment)